dashboard/app/models/pd/application/teacher_application_base.rb

```diff
@@ -65,11 +65,11 @@ def update_user_school_info!
     end
 
     def update_scholarship_status(scholarship_status)
-      Pd::ScholarshipInfo.update_or_create(user, application_year, scholarship_status)
+      Pd::ScholarshipInfo.update_or_create(user, application_year, course_name, scholarship_status)
```

Comment:
I must say I like the use of keyword arguments in the otherwise-similar-looking call to `Pd::ScholarshipInfo.find_by` immediately below...